src/java.desktop/windows/native/libawt/windows/awt_Win32GraphicsDevice.cpp

```diff
@@ -179,7 +185,13 @@ void AwtWin32GraphicsDevice::Initialize()
     }
     gpBitmapInfo->bmiHeader.biBitCount = 0;
     HDC hBMDC = this->GetDC();
+    if (hBMDC == NULL) {
+        J2dTraceLn(J2D_TRACE_WARNING, "AwtWin32GraphicsDevice::Initialize GetDC failed");
+    }
     HBITMAP hBM = ::CreateCompatibleBitmap(hBMDC, 1, 1);
+    if (hBM == NULL) {
+        J2dTraceLn(J2D_TRACE_WARNING, "AwtWin32GraphicsDevice::Initialize CreateCompatibleBitmap failed");
+    }
```

Comment:
Does it make sense to call `::GetLastError()` and trace its value?

It seems that we cannot continue if `hBMDC` or `hBM` are null. Yet it seems unexpected.

Reply:
> It seems that we cannot continue if `hBMDC` or `hBM` are null. Yet it seems unexpected.

Should we rather modify the code to back out and not to call APIs with NULL handles?

Reply:
I think we should fallback to something. The bug itself looks similar to this: https://bugs.openjdk.org/browse/JDK-8185862

Reply:
> I think we should fallback to something. The bug itself looks similar to this: https://bugs.openjdk.org/browse/JDK-8185862

[JDK-8185862](https://bugs.openjdk.org/browse/JDK-8185862) seems to be the exact duplicate of this one.

It references [JDK-8269529](https://bugs.openjdk.org/browse/JDK-8269529) which implemented [a workaround](https://github.com/openjdk/jdk/commit/d042029509a8cbdb723f78e2cfee4e2885775814#diff-fba4a205d0e60634be77d7a7f16a582277f7461d92a9c70cfb96d4a87068d0f1R36):

https://github.com/openjdk/jdk/blob/d042029509a8cbdb723f78e2cfee4e2885775814/test/jdk/javax/swing/reliability/HangDuringStaticInitialization.java#L36

[The same workaround](https://github.com/openjdk/jdk/commit/5c083e8560ce9cc78615e3149a558206724cff53#diff-3b5e63f29c2a52feda881e4115e5d09d1b8aecd4698f1d72981de94caa3a46d0R50) was implemented for [JDK-8266129](https://bugs.openjdk.org/browse/JDK-8266129):

https://github.com/openjdk/jdk/blob/5c083e8560ce9cc78615e3149a558206724cff53/test/jdk/tools/jpackage/windows/WinInstallerIconTest.java#L50

There's @prrace's comment [in the JBS](https://bugs.openjdk.org/browse/JDK-8266129?focusedId=14416854&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14416854): <q cite="https://bugs.openjdk.org/browse/JDK-8266129?focusedId=14416854&page=com.atlassian.jira.plugin.system.issuetabpanels:comment-tabpanel#comment-14416854">There's a bunch of AWT implementation missing to properly work headless. Random tests run into it. The tests should not have to deal with it. Instead it requires an unfunded effort to rework headless mode support in AWT and 2D.</q>

So, we can add `@requires !vm.debug` to the `MultiResolutionImageObserverTest.java` test or, better, handle the situation gracefully.